### DISCUSSION TITLE
Understanding the loading state

### DISCUSSION THREAD

**Discussion by Ben52** [Akryum/vue-apollo#217] [Q&A]
There seem to be two different loading states that are not connected, unless I am misunderstanding something. When a query result is received, the `result` function is called with the data and a loading argument. I have not seen that ever set to true, because by the time the result function is called, its no longer loading. However, there is also the function `watchLoading` which is called immediately, and passed the loading state, which is initially true, and called again when the result is received, with false. The `$apollo.loading` property seems to be using that as well. The ApolloQuery component passes back a loading prop back, which comes from the result function, not the `watchLoading` function, which is called after the result is received. If anyone can help me understand this, it would be appreciated.

**Comment by ghost** [comment]
I am a bit confused on this one as well. It's never changing for me (I am not making any queries so far, only mutations).
Is this expected behavior?

**Comment by zzal** [comment]
With the ApolloQuery component, whenever the `networkStatus` changes, the `@result` event is re-emitted, so the 1st time `loading` should be `=== true`.
That’s my understanding and I’ve tried it in my code and that’s the behavior I get.
The thing I don’t understand is the `:notify-on-network-status-change=""` property, I could not find any difference whether it is set to true or false. Unless it is dispatching via another event, or something…

**Comment by bbugh** [comment]
The network status change notification may have been fixed recently: 

https://github.com/Akryum/vue-apollo/commit/ddce641cad1fddbcfd6bcf4f1f6e3c19826fe246

**Comment by rnenjoy** [comment]
I'm using nuxt and this.$apollo.loading is always false. Never changes during anything. ANd ive tried every cache setting there is i think.
